### ISSUE TITLE
some tutorial models are taking a really long time to fit

### ISSUE THREAD

**Issue by martinamorris** [statnet/tergm#31]
this one in particular

```
data(florentine)
X11()
tergm.fit.1 <- tergm(
  flobusiness ~ 
    Form(~edges+gwesp(0,fixed=T)) + 
    Diss(~offset(edges)),
  targets = "formation",
  offset.coef = log(9),
  estimate = "EGMME",
  control = control.tergm(SA.plot.progress=TRUE)
  )
dev.off()
```

@pavel, this seems to be another example of `gwesp(0,fixed=T)` maybe causing problems, like we saw with `ergm.ego` yesterday.

**Comment by krivit** [comment]
Please stop harassing GitHub user `@pavel` (who happens to not be me). :-)

Regarding the issue, I'll look into it, but I believe it was the `fixed=FALSE` case that caused the real problem. `fixed=TRUE` was just slow.

**Comment by martinamorris** [comment]
lol, sorry about the unintended harassment.

the output suggests it's "Phase 3" that is taking the most time.  should it be taking this long?

```
library(tergm)

data(florentine)
X11()
tergm.fit.1 <- tergm(
  flobusiness ~ 
    Form(~ edges + gwesp(0, fixed=T)) + 
    Diss(~ offset(edges)),
  targets = "formation",
  offset.coef = log(9),
  estimate = "EGMME",
  control = control.tergm(SA.plot.progress=TRUE)
  )

...
========  Phase 1: Burn in, get initial gradient values, and find a configuration under which all targets vary. ========
Burning in... Done.
Attempt 1 :
All parameters have some effect and all statistics are moving. Proceeding to Phase 2.
========  Phase 2: Find and refine the estimate. ========
Subphase 2.1 ////////////////\\\\\
Subphase 2.2 \\\\\
Subphase 2.3 \\\\\
Subphase 2.4 \\\\\
========  Phase 3: Simulate from the fit and estimate standard errors. ========
Subphase 2.5 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////\///\\\\/\\\\\
========  Phase 3: Simulate from the fit and estimate standard errors. ========
Subphase 2.6 \\\\\
========  Phase 3: Simulate from the fit and estimate standard errors. ========

**Comment by krivit** [comment]
Runs pretty fast for me:
``` r
# remotes::install_github("krivit/msgtime") # Run the first time
library(tergm)
#> [startup messages]
data(florentine)
X11()
msgtime::msgtime(TRUE)
tergm.fit.1 <- tergm(
  flobusiness ~ 
    Form(~ edges + gwesp(0, fixed=T)) + 
    Diss(~ offset(edges)),
  targets = "formation",
  offset.coef = log(9),
  estimate = "EGMME",
  control = control.tergm(SA.plot.progress=TRUE)
  ); message("Done!")
#> [2021-06-06 08:45:57]
#> Starting maximum pseudolikelihood estimation (MPLE):
#> [2021-06-06 08:45:57] Evaluating the predictor and response matrix.
#> [2021-06-06 08:45:57] Maximizing the pseudolikelihood.
#> [2021-06-06 08:45:57] Finished MPLE.
#> [2021-06-06 08:45:57] Starting Monte Carlo maximum likelihood estimation (MCMLE):
#> [2021-06-06 08:45:57] Iteration 1 of at most 60:
#> [2021-06-06 08:45:58] Optimizing with step length 1.0000.
#> [2021-06-06 08:45:58] The log-likelihood improved by 0.7097.
#> [2021-06-06 08:45:58] Estimating equations are not within tolerance region.
#> [2021-06-06 08:45:58] Iteration 2 of at most 60:
#> [2021-06-06 08:45:58] Optimizing with step length 1.0000.
#> [2021-06-06 08:45:58] The log-likelihood improved by 0.1446.
#> [2021-06-06 08:45:58] Estimating equations are not within tolerance region.
#> [2021-06-06 08:45:58] Iteration 3 of at most 60:
#> [2021-06-06 08:45:58] Optimizing with step length 1.0000.
#> [2021-06-06 08:45:58] The log-likelihood improved by 0.0030.
#> [2021-06-06 08:45:58] Convergence test p-value: 0.0868. [2021-06-06 08:45:58] Not converged with 99% confidence; increasing sample size.
#> [2021-06-06 08:45:58] Iteration 4 of at most 60:
#> [2021-06-06 08:45:58] Optimizing with step length 1.0000.
#> [2021-06-06 08:45:58] The log-likelihood improved by 0.0142.
#> [2021-06-06 08:45:58] Convergence test p-value: 0.0347. [2021-06-06 08:45:58] Not converged with 99% confidence; increasing sample size.
#> [2021-06-06 08:45:58] Iteration 5 of at most 60:
#> [2021-06-06 08:45:58] Optimizing with step length 1.0000.
#> [2021-06-06 08:45:58] The log-likelihood improved by 0.1106.
#> [2021-06-06 08:45:58] Estimating equations are not within tolerance region.
#> [2021-06-06 08:45:58] Estimating equations did not move closer to tolerance region more than 1 time(s) in 4 steps; increasing sample size.
#> [2021-06-06 08:45:58] Iteration 6 of at most 60:
#> [2021-06-06 08:45:58] Optimizing with step length 1.0000.
#> [2021-06-06 08:45:59] The log-likelihood improved by 0.0585.
#> [2021-06-06 08:45:59] Convergence test p-value: 0.0096. [2021-06-06 08:45:59] Converged with 99% confidence.
#> [2021-06-06 08:45:59] Finished MCMLE.
#> [2021-06-06 08:45:59] This model was fit using MCMC.  To examine model diagnostics and check
#> for degeneracy, use the mcmc.diagnostics() function.
#> Warning in do.call(backup$warning, args): [2021-06-06 08:45:59] Progress plot
#> requested on a non-interactive graphics device. Ignoring.
#> [2021-06-06 08:45:59] ========  Phase 1: Burn in, get initial gradient values, and find a configuration under which all targets vary. ========
#> [2021-06-06 08:45:59] Burning in... [2021-06-06 08:46:00] Done.
#> [2021-06-06 08:46:00] Attempt 1 :
#> [2021-06-06 08:46:07] All parameters have some effect and all statistics are moving. Proceeding to Phase 2.
#> [2021-06-06 08:46:07] ========  Phase 2: Find and refine the estimate. ========
#> [2021-06-06 08:46:07] Subphase 2.1 [2021-06-06 08:46:08] /[2021-06-06 08:46:09] /[2021-06-06 08:46:11] /[2021-06-06 08:46:12] /[2021-06-06 08:46:13] /[2021-06-06 08:46:14] /[2021-06-06 08:46:15] /[2021-06-06 08:46:16] /[2021-06-06 08:46:17] /[2021-06-06 08:46:19] /[2021-06-06 08:46:20] /[2021-06-06 08:46:21] /[2021-06-06 08:46:22] /[2021-06-06 08:46:23] \[2021-06-06 08:46:24] \[2021-06-06 08:46:25] \[2021-06-06 08:46:27] \[2021-06-06 08:46:28] \[2021-06-06 08:46:28] 
#> [2021-06-06 08:46:28] Subphase 2.2 [2021-06-06 08:46:30] /[2021-06-06 08:46:31] \[2021-06-06 08:46:32] \[2021-06-06 08:46:33] \[2021-06-06 08:46:34] \[2021-06-06 08:46:36] \[2021-06-06 08:46:36] 
#> [2021-06-06 08:46:36] Subphase 2.3 [2021-06-06 08:46:37] \[2021-06-06 08:46:39] \[2021-06-06 08:46:40] \[2021-06-06 08:46:41] \[2021-06-06 08:46:42] \[2021-06-06 08:46:42] 
#> [2021-06-06 08:46:42] Subphase 2.4 [2021-06-06 08:46:43] \[2021-06-06 08:46:44] \[2021-06-06 08:46:45] \[2021-06-06 08:46:46] \[2021-06-06 08:46:47] \[2021-06-06 08:46:47] 
#> [2021-06-06 08:46:47] ========  Phase 3: Simulate from the fit and estimate standard errors. ========
#> [2021-06-06 08:46:51] Done!
msgtime::msgtime(FALSE)
```
<sup>Created on 2021-06-06 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

So, a bit under a minute for me for the whole fit, of which Phase 3 takes up about 4 seconds.

**Comment by martinamorris** [comment]
running multiple times i get a range of 1-3 minutes.  the longer times have a long phase 3.

**Comment by chad-klumb** [comment]
this was addressed with 72080206c25ae07e0135d715a623c25d1aca7fb6
